src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/MessagePageRenderer.cs

```diff
@@ -0,0 +1,166 @@
+﻿using Android.Content;
+using Android.Runtime;
+using Android.Views;
+using Android.Webkit;
+using Android.Widget;
+using UrbanAirship.NETStandard.MessageCenter;
+using UrbanAirship.RichPush;
+using UrbanAirship.Widget;
+using Xamarin.Forms.Platform.Android;
+
+namespace UrbanAirship.NETStandard.Android
+{
+    public class MessagePageRenderer : PageRenderer, RichPushInbox.IFetchMessagesCallback
+    {
+        private class MessageWebViewClient : UAWebViewClient
+        {
+            private MessagePageRenderer renderer;
+            private bool error = false;
+
+            public MessageWebViewClient(MessagePageRenderer messagePageRenderer)
+            {
+                renderer = messagePageRenderer;
+            }
+
+            public override void OnPageFinished(WebView view, string url)
+            {
+                base.OnPageFinished(view, url);
+
+                if (renderer.message == null)
+                {
+                    return;
+                }
+
+                if (error)
+                {
+                    renderer.messagePage.OnRendererLoadFailed(renderer.message.MessageId, false, MessageFailureStatus.LoadFailed);
+                    return;
+                }
+
+                renderer.message.MarkRead();
+                renderer.messagePage.OnRendererLoaded(renderer.message.MessageId);
+            }
+
+#pragma warning disable CS0672 // Member overrides obsolete member
```

Comment:
these warnings are about deprecated methods we still have to use if we're targeting < API 23